rust/xaynet-server/src/rest.rs

```diff
@@ -1,22 +1,29 @@
 //! A HTTP API for the PET protocol interactions.
 
-use crate::services::{fetchers::Fetcher, messages::PetMessageHandler};
-use bytes::{Buf, Bytes};
 use std::{convert::Infallible, net::SocketAddr};
+
+use bytes::{Buf, Bytes};
 use warp::{
     http::{Response, StatusCode},
     Filter,
 };
+
+use crate::{
+    services::{fetchers::Fetcher, messages::PetMessageHandler},
+    settings::TlsSettings,
+};
 use xaynet_core::{crypto::ByteObject, ParticipantPublicKey};
 
 /// Starts a HTTP server at the given address, listening to GET requests for
 /// data and POST requests containing PET messages.
 ///
 /// * `addr`: address of the server.
+/// * `tls_settings`: optional certificate and key for TLS server authentication.
 /// * `fetcher`: fetcher for responding to data requests.
 /// * `pet_message_handler`: handler for responding to PET messages.
 pub async fn serve<F>(
     addr: impl Into<SocketAddr> + 'static,
+    tls_settings: TlsSettings,
```

Comment:
If we put the tls setting into the api setting, we can simplify this part a bit

```
pub async fn serve<F>(
    api_settings: ApiSettings,
    fetcher: F,
    pet_message_handler: PetMessageHandler,
)
```

Reply:
agree.